typescript/sdk/src/token/EvmWarpHyperlaneModule.ts

```diff
@@ -0,0 +1,227 @@
+  ): Promise<EthersV5Transaction[]> {
+    return [
+      ...(await this.updateIsm(config)),
+      ...(await this.updateHook(config)),
+    ];
+  }
+
+  /**
+   * Updates an existing Warp route ISM with a given configuration.
+   *
+   * This method handles two cases:
+   * 1. If the `config.interchainSecurityModule` is a string
+   *  - It attempts to derive the ISM from the provided string, and updates the contract's ISM.
+   * 2. If the `config.interchainSecurityModule` is an object,
+   *  - Checks if the current onchain ISM configuration matches the provided configuration.
+   *  - If not, it deploys a new ISM module, and updates the contract's ISM.
+   *
+   * @param config - The token router configuration, including the ISM configuration.
+   * @returns An array of Ethereum transactions that need to be executed to update the ISM configuration.
+   */
+  async updateIsm(
+    config: DerivedTokenRouterConfig,
+  ): Promise<EthersV5Transaction[]> {
+    const transactions: EthersV5Transaction[] = [];
+    if (config.interchainSecurityModule) {
+      const contractToUpdate = await this.args.addresses[
+        config.type
+      ].deployed();
+
+      // Attempt to derive the config.interchainSecurityModule if its a string, or deploy it if its an IsmConfig
+      const expectedIsm =
+        typeof config.interchainSecurityModule === 'string'
+          ? await this.reader.evmIsmReader.deriveIsmConfig(
+              config.interchainSecurityModule,
+            )
+          : await this.deployIsm(config);
+
+      // read() will return undefined if no ISM is set, so make it an IsmConfig-ish object
+      const actualIsm = (await this.read()).interchainSecurityModule ?? {
+        address: ethers.constants.AddressZero,
+      };
+
+      if (!deepEquals(expectedIsm, actualIsm)) {
+        transactions.push({
+          transaction:
+            await contractToUpdate.populateTransaction.setInterchainSecurityModule(
+              expectedIsm.address,
+            ),
+          type: ProviderType.EthersV5,
+        });
+      }
+    }
+
+    return transactions;
+  }
+
+  /**
+   * Deploys the ISM using the provided configuration.
+   *
+   * @param config - The configuration for the ISM to be deployed.
+   * @returns The deployed ISM contract address.
+   */
+  public async deployIsm(
+    config: DerivedTokenRouterConfig,
+  ): Promise<DerivedIsmConfigWithAddress> {
+    // Take the config.ismFactoryAddresses, de-serialize them into Contracts, and pass into EvmIsmModule.create
+    const factories = attachContracts(
+      config.ismFactoryAddresses as HyperlaneAddresses<ProxyFactoryFactories>,
+      proxyFactoryFactories,
+    );
+    const evmIsmModule = await EvmIsmModule.create({
+      chain: this.args.chain,
+      config: config.interchainSecurityModule!,
+      deployer: new HyperlaneProxyFactoryDeployer(this.multiProvider),
+      factories,
+      multiProvider: this.multiProvider,
+    });
+    return evmIsmModule.read();
```

Comment:
dont really follow why we are reading in here?

Reply:
Update to return the config.hook with deployedIsm address attached to reduce `read()` calls